beancount/loader_test.py

```diff
@@ -559,7 +579,8 @@ def test_load_cache(self):
                 },
             )
             top_filename = path.join(tmp, "apples.beancount")
-            entries, errors, options_map = loader.load_file(top_filename)
+            with mock_time_time():
+                entries, errors, options_map = loader.load_file(top_filename)
```

Comment:
I'm a bit suspicious of this change: I don't think that this code has always been run on system where it takes more than 1.2 seconds to execute. Or is it the other way around and it takes more than 1.2 seconds to execute it on the Windows CI machines and you are artificially capping the execution time to 1.2 seconds? I agree that having the test depend on the execution time is not good, but this deserves a comment explaining what is going on.

Reply:
Looks like it's the other way around

https://github.com/beancount/beancount/blob/793d8be9f3e7f47230c270d6f0faf015ccf4cdab/beancount/loader.py#L236-L238

We only cache beancount take more than 1s to load, and we have test to check if cache is generated and tests pass on POSIX and windows ci(py<3.13).

Another options here is to mock the real loader and add a `time.sleep` in it.

Reply:
I still don't get it. Are you saying that the test works only if the loafing takes more than 1 seconds? I'm pretty sure that it takes less than than on my machine.

Reply:
> I still don't get it. Are you saying that the test works only if the loafing takes more than 1 seconds? I'm pretty sure that it takes less than than on my machine.

I need to take a deeper look into it, I guess it also has something to do with the time resolution.

Reply:
OK, I know. on windows the time resolution is not small enough to distinguish time_before and time_after, so even we have `PICKLE_CACHE_THRESHOLD=0` we will need to mock time.

Reply:
and we should use `time.monotonic()` here, even though it can't avoid time resolution problem.

Reply:
Somewhere there must be a condition equivalent to `(time_after - time_before) >PICKLE_CACHE_THRESHOLD` which fails when `time_after == time_before`. What about turning `>` into `>=`? That should solve the issue without any practical consequence on real usage (on Windows the default resolution for `time.time()` is 13 ms and `PICKLE_CACHE_THRESHOLD` is much larger than than in any practical scenario. Would this fix the issue?

Reply:
Namely, turn this line https://github.com/beancount/beancount/blob/793d8be9f3e7f47230c270d6f0faf015ccf4cdab/beancount/loader.py#L242
into
```python
    if time_after - time_before >= time_threshold:
```

Reply:
> Somewhere there must be a condition equivalent to `(time_after - time_before) >PICKLE_CACHE_THRESHOLD` which fails when `time_after == time_before`. What about turning `>` into `>=`? That should solve the issue without any practical consequence on real usage (on Windows the default resolution for `time.time()` is 13 ms and `PICKLE_CACHE_THRESHOLD` is much larger than than in any practical scenario. Would this fix the issue?

yes, that's one option

Reply:
Another option is to use an higher resolution clock via `time.perf_counter()`.

Reply:
that's `time.monotonic`